### PR TITLE
Bug fixes : Theano shape inference

### PR DESCRIPTION
**Pull request by farizrahman4u**

### CODE REVIEW

**Comment by farizrahman4u** [comment]
@abhaikollara Please review.

**Comment by abhaikollara** [comment]
@farizrahman Will do

**Comment by farizrahman4u** [comment]
@fchollet This is important.

**Comment by abhaikollara** [comment]
@farizrahman4u I'll add some tests soon

**Comment by farizrahman4u** [comment]
@abhaikollara PR to this branch (farizrahman4u/keras:bug-fix)

File: keras/backend/theano_backend.py [hunk 1]
```diff
@@ -821,22 +822,37 @@ def arange(start, stop=None, step=1, dtype='int32'):
...
+            if x._keras_shape[-1] is None:
+                output_shape += (None,)
+            else:
+                output_shape += (x._keras_shape[-1] * n,)
         else:
-            xshape = np.append([1] * -diff, xshape)
-        y._keras_shape = tuple(xshape * n)
-
+            # symbolic n
+            n_ndim = K.ndim(n)
```
Comment (abhaikollara) [inline keras/backend/theano_backend.py]: @farizrahman4u Undefined variable K

Comment (abhaikollara) [inline keras/backend/theano_backend.py]: Isn't `ndim` always going to be 1 for `n`, I think you meant the equivalent of `len(n)`

**Comment by farizrahman4u** [comment]
@abhaikollara Unit tests?

**Comment by fchollet** [comment]
Should I merge or are there incoming tests? In any case, there should be unit tests.

**Comment by farizrahman4u** [comment]
@fchollet Tests added.

**Comment by abhaikollara** [comment]
@farizrahman4u Sorry about the unit tests, wasn't feeling well enough to code much. Should I add tests for `batch_flatten` ?

**Comment by farizrahman4u** [comment]
Yeah, for `flatten` too, if possible.

**Comment by abhaikollara** [comment]
@farizrahman4u I've made a PR

**Comment by fchollet** [comment]
LGTM, thanks.

**Pull request merged**